backbone.js

```diff
@@ -171,48 +130,33 @@
     // listen to an event in another object ... keeping track of what it's
     // listening to.
     listenTo: function(obj, name, callback) {
-      var listeningTo = this._listeningTo || (this._listeningTo = {});
+      if (!obj) return this;
       var id = obj._listenId || (obj._listenId = _.uniqueId('l'));
-      listeningTo[id] = obj;
-      if (!callback && typeof name === 'object') callback = this;
+      var listeningTo = this._listeningTo || (this._listeningTo = {});
+      var listening = listeningTo[id];
+
+      if (!listening) {
+        listening = listeningTo[id] = {obj: obj, events: {}};
+        id = this._listenId || (this._listenId = _.uniqueId('l'));
+        var listeners = obj._listeners || (obj._listeners = {});
+        listeners[id] = this;
+      }
+
+      // Bind callbacks on obj, and keep track of them on listening.
       obj.on(name, callback, this);
+      listening.events = eventsApi(onApi, listening.events, name, callback);
       return this;
     },
 
     listenToOnce: function(obj, name, callback) {
-      if (typeof name === 'object') {
-        for (var event in name) this.listenToOnce(obj, event, name[event]);
-        return this;
-      }
-      if (eventSplitter.test(name)) {
-        var names = name.split(eventSplitter);
-        for (var i = 0, length = names.length; i < length; i++) {
-          this.listenToOnce(obj, names[i], callback);
-        }
-        return this;
-      }
-      if (!callback) return this;
-      var once = _.once(function() {
-        this.stopListening(obj, name, once);
-        callback.apply(this, arguments);
-      });
-      once._callback = callback;
-      return this.listenTo(obj, name, once);
+      name = onceMap(name, callback, _.bind(this.stopListening, this, obj));
```

Comment:
Can't, because `name` could be an object or space-separated string. This will be passed in the normalized (just a single event name and it's callback) by the [onceWrapper](https://github.com/jashkenas/backbone/pull/3455/files#diff-0d56d0d310de7ff18b3cef9c2f8f75dcR286).